src/test/java/seedu/academyassist/logic/parser/ParserUtilTest.java

```diff
@@ -149,48 +147,51 @@ public void parseEmail_validValueWithWhitespace_returnsTrimmedEmail() throws Exc
     }
 
     @Test
-    public void parseTag_null_throwsNullPointerException() {
-        assertThrows(NullPointerException.class, () -> ParserUtil.parseTag(null));
+    public void parseSubject_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> ParserUtil.parseSubject(null));
     }
 
     @Test
-    public void parseTag_invalidValue_throwsParseException() {
-        assertThrows(ParseException.class, () -> ParserUtil.parseTag(INVALID_TAG));
+    public void parseSubject_invalidValue_throwsParseException() {
+        assertThrows(ParseException.class, () -> ParserUtil.parseSubject(INVALID_SUBJECT));
     }
 
     @Test
-    public void parseTag_validValueWithoutWhitespace_returnsTag() throws Exception {
-        Tag expectedTag = new Tag(VALID_TAG_1);
-        assertEquals(expectedTag, ParserUtil.parseTag(VALID_TAG_1));
+    public void parseSubject_validValueWithoutWhitespace_returnsSubject() throws Exception {
+        Subject expectedSubject = new Subject(VALID_SUBJECT_1);
+        assertEquals(expectedSubject, ParserUtil.parseSubject(VALID_SUBJECT_1));
     }
 
     @Test
-    public void parseTag_validValueWithWhitespace_returnsTrimmedTag() throws Exception {
-        String tagWithWhitespace = WHITESPACE + VALID_TAG_1 + WHITESPACE;
-        Tag expectedTag = new Tag(VALID_TAG_1);
-        assertEquals(expectedTag, ParserUtil.parseTag(tagWithWhitespace));
+    public void parseSubject_validValueWithWhitespace_returnsTrimmedSubject() throws Exception {
+        String subjectWithWhitespace = WHITESPACE + VALID_SUBJECT_1 + WHITESPACE;
+        Subject expectedSubject = new Subject(VALID_SUBJECT_1);
+        assertEquals(expectedSubject, ParserUtil.parseSubject(subjectWithWhitespace));
     }
 
     @Test
-    public void parseTags_null_throwsNullPointerException() {
-        assertThrows(NullPointerException.class, () -> ParserUtil.parseTags(null));
+    public void parseSubjects_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> ParserUtil.parseSubjects(null));
     }
 
     @Test
-    public void parseTags_collectionWithInvalidTags_throwsParseException() {
-        assertThrows(ParseException.class, () -> ParserUtil.parseTags(Arrays.asList(VALID_TAG_1, INVALID_TAG)));
+    public void parseSubjects_collectionWithInvalidSubjects_throwsParseException() {
+        assertThrows(ParseException.class, () -> ParserUtil.parseSubjects(Arrays.asList(VALID_SUBJECT_1,
+                INVALID_SUBJECT)));
     }
 
+    // todo check the below
     @Test
-    public void parseTags_emptyCollection_returnsEmptySet() throws Exception {
-        assertTrue(ParserUtil.parseTags(Collections.emptyList()).isEmpty());
+    public void parseSubjects_emptyCollection_throwsParseException() throws Exception {
+        assertThrows(ParseException.class, () -> ParserUtil.parseSubjects(Collections.emptyList()).isEmpty());
```

Comment:
Perhaps remove `isEmpty()`? It seems unnecessary since it won't execute if `parseSubjects` throws `ParseException`.